tfjs-backend-webgpu/perf/tune.js

```diff
@@ -0,0 +1,381 @@
+  }
+  const scaleSelected = [];
+  const iterationSelected = [];
+
+  // clear compared result
+  webglCompResults = [];
+
+  // define scale suite
+  let scales = document.querySelectorAll('.scaleCheckBox:checked');
+  for (let s of scales) {
+    scaleSelected.push(s.value);
+  }
+
+  // define iteration suite
+  let iterations = document.querySelectorAll('.iterationCheckBox:checked');
+  for (let i of iterations) {
+    iterationSelected.push(i.value);
+  }
+
+  INPUTS = customInputs;
+  for (let scale of scaleSelected) {
+    let mySet = getTestSet(scale);
+    INPUTS = INPUTS.concat([...mySet]);
+    for (let input of INPUTS) {
+      for (let iteration of iterationSelected) {
+        STATE.input = input;
+        STATE.iteration = iteration;
+        await timeMatmul();
+      }
+    }
+  }
+  document.getElementById('message').innerHTML = 'Done!';
+  updateColor();
+}
+
+async function rerun(rowNum) {
+  let iteration = document.getElementById(`iteration-${rowNum}`).getAttribute('value');
+  let input = document.getElementById(`input-${rowNum}`).getAttribute('value');
+  STATE.input = input;
+  STATE.iteration = iteration;
+  await timeMatmul(rowNum);
+  document.getElementById('message').innerHTML = 'Done!';
+  updateColor();
+}
+
+const getCellValue = (tr, idx) => tr.children[idx].innerText || tr.children[idx].textContent;
+
+const comparer = (idx, asc) => (a, b) => ((v1, v2) =>
+  v1 !== '' && v2 !== '' && !isNaN(v1) && !isNaN(v2) ? v1 - v2 : v1.toString().localeCompare(v2)
+)(getCellValue(asc ? a : b, idx), getCellValue(asc ? b : a, idx));
+
+function getFactors(c) {
+  const y = [];
+  let min = 1;
+  let max = c;
+  while (min < max && min * min <= c) {
+    if (c % min === 0) {
+      y.push(min, c / min);
+      max = c / min;
+    }
+    min++;
+  }
+  return y;
+}
+
+function getTestSet(num) {
+  const factors = getFactors(num);
+  const mySet = new Set();
+  for (let i = 0; i < factors.length; i++) {
+    for (let j = 0; j < factors.length; j++) {
+      if (factors[i] * factors[j] < num && num % (factors[i] * factors[j]) === 0) {
+        let m = factors[i];
+        let n = factors[j];
+        let k = num / factors[i] / factors[j];
+        if (m > 15 && n > 15 && k > 15
+          && n < 2048 && k < 2048
+          && Math.sqrt(m) % 1 === 0) {
+          mySet.add(`${m},${k},${n}`);
+        }
+      }
+    }
+  }
+  return mySet;
+}
+
+function getPrimeFactor(num) {
+  const prime_factor = [];
+  for (let i = 2; i < num; i++) {
+    if (num % i == 0) {
+      prime_factor.push(i);
+      num /= i;
+      i -= 1;
+    }
+  }
+  prime_factor.push(num);
+  return prime_factor;
+}
+
+const STATE = {
+  name: 'matMul',
+  backend: 'webgpu',
+  input: '144,256,256',
+  iteration: 500,
+};
+
+const resultMap = new Map();
+let webglCompResults = [];
+
+let SCALES = [1016064, 5013504, 10838016, 33554432];
+let currentScale = [1016064];
+let customInputs = ['720, 512, 1536', '1, 704, 2000', '1, 128, 1404', '1, 1280, 1001'];
```

Comment:
It seems that the customInputs and the other test set are run together. There is no place to change the customInputs or only run the customInputs.